src/Miki/Services/Settings/SettingsService.cs

```diff
@@ -0,0 +1,51 @@
+﻿namespace Miki.Services.Settings
+{
+    using System;
+    using System.Threading.Tasks;
+    using Miki.Bot.Models;
+    using Miki.Bot.Models.Exceptions;
+    using Miki.Framework;
+    using Miki.Functional;
+    using Miki.Patterns.Repositories;
+
+    public class SettingsService : ISettingsService
+    {
+        private readonly IUnitOfWork unit;
+        private readonly IAsyncRepository<Setting> settingsRepository;
+
+        public async ValueTask<Result<T>> GetAsync<T>(SettingType type, long entityId)
+            where T : Enum
+        {
+            var setting = await settingsRepository.GetAsync(entityId, (int)type);
+            if(setting == null)
+            {
+                return new Result<T>(
+                    new EntityNullException<Setting>());
+            }
+            return (T)(object)setting.Value;
```

Comment:
Could be improved by using omitting boxing, but there's no easy way for now to do it.

Reply:
Add a todo or an issue for it